index.js

```diff
@@ -74,71 +81,97 @@ module.exports = function (homebridge) {
     }
 
     handleMessage(json) {
+      let item = null;
+
       if (utils.isMessageOfTypeValues(json)) {
         // bulk update ("request values")
-        const item = json.find((item) => {
+        item = json.find((item) => {
           return item.devices.indexOf(this.config.deviceId) !== -1;
         });
-        if (item) {
-          switch (this.config.type) {
-            case "TemperatureSensor":
-              this.deviceState = item.values.temperature;
-              this.log(`Initialized device with temperature "${item.values.temperature}"`);
-              break;
-            default:
-              // or Switch
-              this.deviceState = item.values.state === 'on';
-              this.log(`Initialized device with state "${item.values.state}"`);
-              break;
-          }
-        } else {
-          this.log(`Could not find device with id "${this.config.deviceId}"`);
-        }
       } else if (utils.isMessageOfTypeUpdate(json)) {
         // item update (after "control")
         if (json.devices.indexOf(this.config.deviceId) !== -1) {
-          let characteristic = "";
-          switch (this.config.type) {
-            case "TemperatureSensor":
-              characteristic = homebridge.hap.Characteristic.CurrentTemperature;
-              this.deviceState = json.values.temperature;
-              this.log(`Updated internal state to "${json.values.temperature}"`);
-              break;
-            default:
-              // or Switch
-              characteristic = homebridge.hap.Characteristic.On;
-              this.deviceState = json.values.state === 'on';
-              this.log(`Updated internal state to "${json.values.state}"`);
-              break;
-          }
-
-          //Trigger an update to Homekit
```

Comment:
Thanks for pointing me to the PR this was added in.
I removed it as it was causing problems with the dimmers I explained. But I think I might be able to add it back with some of the other changes I made. I'll report back when I've been able to test it.